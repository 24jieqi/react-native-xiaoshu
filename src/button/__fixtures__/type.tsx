/**
 * title: 按钮类型
 * description: 按钮有五种类型：主按钮（默认）、朦胧按钮、边框按钮、幽灵按钮和链接按钮。
 */

import React, { memo } from 'react'

import {
  Button,
  Card,
  Space,
  Divider,
} from '@fruits-chain/react-native-xiaoshu'

const onPress = () => {
  console.log('点击按钮 => ', new Date().getTime())
}

const ButtonType: React.FC = () => {
  return (
    <Card title="按钮类型" square>
      <Space>
        <Space>
          <Button
            type="primary"
            text="primary 主按钮&点击事件防抖动"
            onPress={onPress}
            onPressDebounceWait={500}
          />
          <Button
            type="hazy"
            text="hazy 朦胧按钮"
            onPress={onPress}
            onPressDebounceWait={300}
          />
          <Button type="outline" text="outline 边框按钮" onPress={onPress} />
          <Button type="ghost" text="ghost 幽灵按钮" onPress={onPress} />
          <Button type="link" text="link 链接按钮" onPress={onPress} />
        </Space>

        <Divider>·</Divider>

        <Space direction="horizontal" wrap>
          <Button type="primary" text="primary" onPress={onPress} />
          <Button type="hazy" text="hazy" onPress={onPress} />
          <Button type="outline" text="outline" onPress={onPress} />
          <Button type="ghost" text="ghost" onPress={onPress} />
          <Button type="link" text="link" onPress={onPress} />
        </Space>
      </Space>
    </Card>
  )
}

export default memo(ButtonType)
