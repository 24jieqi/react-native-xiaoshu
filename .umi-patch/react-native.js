export * from 'react-native-web'

export const InputAccessoryView = () => {}
