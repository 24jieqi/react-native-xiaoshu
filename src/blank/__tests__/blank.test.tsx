import TOKENS from '@fruits-chain/design-tokens-bailu'
import React from 'react'
import type { ViewStyle } from 'react-native'
import { Text, StyleSheet } from 'react-native'
import { create } from 'react-test-renderer'

import Blank from '..'
import { customRender } from '../../__tests__/test-utils'
import type { BlankProps } from '../interface'

describe('Blank', () => {
  it('render snapshot', () => {
    const tree = create(
      <Blank>
        <Text>Content</Text>
      </Blank>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  const sizes: {
    label: BlankProps['size']
    value: number
  }[] = [
    {
      value: TOKENS.space_3,
      label: 'm',
    },
    {
      value: TOKENS.space_2,
      label: 's',
    },
    {
      value: TOKENS.space_4,
      label: 'l',
    },
  ]

  sizes.forEach(({ value, label }) => {
    it(`${label} size margin is ${value}`, () => {
      const { getByTestId, update } = customRender(
        <Blank testID="blank" size={label}>
          <Text>Content</Text>
        </Blank>,
      )

      const node = getByTestId('blank')
      const nodeStyle = StyleSheet.flatten<ViewStyle>(node.props.style)

      expect(nodeStyle.marginLeft).toEqual(value)
      expect(nodeStyle.marginRight).toEqual(value)
      expect(nodeStyle.marginTop).toEqual(0)
      expect(nodeStyle.marginBottom).toEqual(0)

      update(
        <Blank testID="blank" size={label} top bottom>
          <Text>Content</Text>
        </Blank>,
      )

      const node2 = getByTestId('blank')
      const nodeStyle2 = StyleSheet.flatten<ViewStyle>(node2.props.style)

      expect(nodeStyle2.marginLeft).toEqual(value)
      expect(nodeStyle2.marginRight).toEqual(value)
      expect(nodeStyle2.marginTop).toEqual(value)
      expect(nodeStyle2.marginBottom).toEqual(value)

      update(
        <Blank
          testID="blank"
          size={label}
          top
          bottom
          left={false}
          right={false}>
          <Text>Content</Text>
        </Blank>,
      )

      const node3 = getByTestId('blank')
      const nodeStyle3 = StyleSheet.flatten<ViewStyle>(node3.props.style)

      expect(nodeStyle3.marginLeft).toEqual(0)
      expect(nodeStyle3.marginRight).toEqual(0)
      expect(nodeStyle3.marginTop).toEqual(value)
      expect(nodeStyle3.marginBottom).toEqual(value)
    })
  })

  it('custom size', () => {
    const { getByTestId, update } = customRender(
      <Blank testID="blank">
        <Text>Content</Text>
      </Blank>,
    )

    const node = getByTestId('blank')
    const nodeStyle = StyleSheet.flatten<ViewStyle>(node.props.style)

    expect(nodeStyle.marginLeft).toEqual(TOKENS.space_3)
    expect(nodeStyle.marginRight).toEqual(TOKENS.space_3)
    expect(nodeStyle.marginTop).toEqual(0)
    expect(nodeStyle.marginBottom).toEqual(0)

    update(
      <Blank testID="blank" left={6} right={10} top={2} bottom={6}>
        <Text>Content</Text>
      </Blank>,
    )

    const node2 = getByTestId('blank')
    const nodeStyle2 = StyleSheet.flatten<ViewStyle>(node2.props.style)

    expect(nodeStyle2.marginLeft).toEqual(6)
    expect(nodeStyle2.marginRight).toEqual(10)
    expect(nodeStyle2.marginTop).toEqual(2)
    expect(nodeStyle2.marginBottom).toEqual(6)
  })

  it('custom style', () => {
    const { getByTestId } = customRender(
      <Blank
        testID="blank"
        top={20}
        // eslint-disable-next-line react-native/no-inline-styles
        style={{
          marginTop: 16,
          marginBottom: 8,
        }}>
        <Text>Content</Text>
      </Blank>,
    )

    const node = getByTestId('blank')
    const nodeStyle = StyleSheet.flatten<ViewStyle>(node.props.style)

    expect(nodeStyle.marginLeft).toEqual(TOKENS.space_3)
    expect(nodeStyle.marginRight).toEqual(TOKENS.space_3)
    expect(nodeStyle.marginTop).toEqual(16)
    expect(nodeStyle.marginBottom).toEqual(8)
  })
})
