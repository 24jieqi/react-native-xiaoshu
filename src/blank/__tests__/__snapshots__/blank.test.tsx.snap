// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank render snapshot 1`] = `
<View
  style={
    [
      {
        "marginBottom": 0,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 0,
      },
      undefined,
    ]
  }
>
  <Text>
    Content
  </Text>
</View>
`;
